File: backend/SNS.Application/Discussions/Solutions/Solutions/Contracts/SolutionSummaryDto.cs
using SNS.Application.Profiles.Profiles.Contracts;
using SNS.Domain.Discussions.Solutions.Enums;

namespace SNS.Application.Discussions.Solutions.Solutions.Contracts;

/// <summary>
/// Represents summary information for a proposed solution in list views and feeds.
/// </summary>
/// <param name="Id">The unique identifier of the solution.</param>
/// <param name="ProblemId">The associated problem identifier.</param>
/// <param name="Status">The solution status.</param>
/// <param name="Author">Snapshot of the solution author profile.</param>
/// <param name="UpvotesCount">The count of upvotes.</param>
/// <param name="DownvotesCount">The count of downvotes.</param>
/// <param name="DiscussionsCount">The count of threaded discussion comments.</param>
/// <param name="CreatedAt">The timestamp when the solution was submitted.</param>
/// <param name="UpdatedAt">The timestamp of the last update.</param>
public sealed record SolutionSummaryDto(
    Guid Id,
    Guid ProblemId,
    SolutionStatus Status,
    ProfileSnapshotDto Author,
    int UpvotesCount,
    int DownvotesCount,
    int DiscussionsCount,
    DateTime CreatedAt,
    DateTime UpdatedAt
);

File: SNS.Web/src/app/discussions/solutions/solutions/contracts/solution-summary.dto.ts
import { ProfileSnapshotDto } from '../../../../profiles/profiles/contracts/profile-snapshot.dto';
import { VoteType } from '../../../shared/enums/vote-type.enum';
import { SolutionStatus } from '../../enums/solution-status.enum';
import { SolutionContentBlockDto } from './solution-content-block.dto';

export interface SolutionSummaryDto {
    id: string;
    problemId: string;
    status: SolutionStatus;
    author: ProfileSnapshotDto;
    contentBlocks: SolutionContentBlockDto[];
    upvotesCount: number;
    downvotesCount: number;
    discussionsCount: number;
    currentUserVote?: VoteType | null;
    isUpVotedByCurrentUser?: boolean;
    isDownVotedByCurrentUser?: boolean;
    createdAt: string;
    updatedAt: string;
}
